projects/plugins/crm/includes/ZeroBSCRM.DAL3.Obj.Companies.php

```diff
@@ -2759,8 +2761,22 @@ private function tidy_company($obj=false,$withCustomFields=false){
 
             // if have totals, pass them :)
             if (isset($obj->quotes_total)) $res['quotes_total'] = $obj->quotes_total;
-            if (isset($obj->invoices_total)) $res['invoices_total'] = $obj->invoices_total;
-            if (isset($obj->transactions_total)) $res['transactions_total'] = $obj->transactions_total;
+			if ( isset( $obj->invoices_total ) ) {
+				$company                            = $zbs->DAL->companies->getCompany( $res['id'], array( 'withInvoices' => true ) ); // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+				$deleted_invoice_details            = jetpackCRM_deleted_totals( $company['invoices'] );
+				$res['invoices_total']              = $obj->invoices_total - $deleted_invoice_details['total'];
+				$res['invoices_total_with_deleted'] = $deleted_invoice_details['total'];
+				$res['invoices_count_with_deleted'] = $zbs->DAL->customer_has_count_obj_type( $res['id'], ZBS_TYPE_INVOICE, ZBS_TYPE_COMPANY ); // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+				$res['invoices_count']              = $res['invoices_count_with_deleted'] - $deleted_invoice_details['count'];
+			}
+			if ( isset( $obj->transactions_total ) ) {
+				$company                                = $zbs->DAL->companies->getCompany( $res['id'], array( 'withTransactions' => true ) ); // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
```

Comment:
I think this isn't needed due to earlier filtering of transactions?

Reply:
I've removed transaction logic (gone back to how it was) for now - counts aren't correct even though the totals seem to be, but this can be something for another issue.